paprika/setup.py

```diff
@@ -31,732 +26,685 @@ def _get_installed_benchmarks():
-            else f"{self.host}.pdb"))
-
-    def parse_yaml(self, installed_benchmarks, guest_orientation):
+        # Extract the host from the full structure.
+        if not host_atom_indices:
+            host_atom_indices = range(len(structure.atoms))
+
+        host_structure = structure[
+            "@" + ",".join(map(lambda x: str(x + 1), host_atom_indices))
+        ]
+
+        # noinspection PyTypeChecker
+        center_of_mass: np.ndarray = pmd.geometry.center_of_mass(
+            host_structure.coordinates, masses=np.ones(len(host_structure.coordinates))
+        )
+
+        # Remove the COM from the host coordinates to make alignment easier.
+        structure.coordinates -= center_of_mass
+        host_structure.coordinates -= center_of_mass
+
+        # Find the principal components of the host, take the two largest, and find
+        # the vector orthogonal to that. Use that vector to align with the z-axis.
+        # This may not generalize to non-radially-symmetric host molecules.
+        inertia_tensor = np.dot(
+            host_structure.coordinates.transpose(), host_structure.coordinates
+        )
+
+        eigenvalues, eigenvectors = np.linalg.eig(inertia_tensor)
+        order = np.argsort(eigenvalues)
+
+        _, axis_2, axis_1 = eigenvectors[:, order].transpose()
+
+        cavity_axis = np.cross(axis_1, axis_2)
+
+        # Add dummy atoms which will be used to align the structure.
+        cls.add_dummy_atoms_to_structure(structure, [np.array([0, 0, 0]), cavity_axis])
+
+        # Give atoms uniform mass so that the align code uses the center
+        # of geometry rather than the center of mass.
+        for atom in structure.atoms:
+            atom.mass = 1.0
+
+        aligned_structure = align.zalign(structure, ":DM1", ":DM2")
+
+        # Return a copy of the original structure where things like the masses
+        # have not been changed and dummy atoms not added.
+
+        # noinspection PyTypeChecker
+        structure: pmd.Structure = pmd.load_file(coordinate_path, structure=True)
+        structure.coordinates = aligned_structure["!:DM1&!:DM2"].coordinates
+
+        return structure
+
+    @classmethod
+    def prepare_complex_structure(
+        cls,
+        coordinate_path: str,
+        guest_atom_indices: List[int],
+        guest_orientation_mask: str,
+        pull_distance: float,
+        pull_window_index: int,
+        n_pull_windows: int,
+    ) -> pmd.Structure:
+        """Prepares the coordinates of both the host and the guest molecule in a
+        given APR window.
+
+        Parameters
+        ----------
+        coordinate_path
+            The path to the coordinate file which contains the guest molecule
+            bound to the host.
+        guest_atom_indices
+        guest_orientation_mask
+        pull_distance
+        pull_window_index
+        n_pull_windows
```

Comment:
We probably should document these before merging.

Reply:
Added.